client/reader/discover/controller.js

```diff
@@ -70,11 +86,25 @@ const exported = {
 				/>
 			</>
 		);
-		/* eslint-enable wpcalypso/jsx-classname-namespace */
+		next();
+	},
+
+	discoverSsr( context, next ) {
+		context.renderHeaderSection = renderHeaderSection;
+
+		const selectedTab = context.query.selectedTab || DEFAULT_TAB;
```

Comment:
[The SSR is skipped when there's a query strings](https://github.com/Automattic/wp-calypso/blob/c9467f7afcc777c2850861e67f9c8bf10cdbf2c5/client/server/render/index.js#L304), but it should be fine to leave it as is, in case this behavior is updated in the future.

Reply:
Thanks for the link to the source! I did notice the behavior too. I also think it's fine as-is — with the current changes, at least the main Discover page will be indexed nicely, which will probably have the biggest impact.